src/server.rs

```diff
@@ -599,6 +693,43 @@ impl LanguageServer for LspServer {
         }
         Ok(None)
     }
+    async fn rename(
+        &self,
+        params: lsp::RenameParams,
+    ) -> Result<Option<lsp::WorkspaceEdit>> {
+        let key = params.text_document_position.text_document.uri;
+        let store = self.store.lock().unwrap();
+        if !store.contains_key(&key) {
+            error!(
+                "textDocument/didChange called on unknown file {}",
+                key
+            );
+        }
+        let contents = store.get(&key).unwrap();
+        let pos = params.text_document_position.position;
+        let pkg = parse_and_analyze(contents);
+        let node = find_node(walk::Node::Package(&pkg), pos);
+
+        let locations = find_references(key.clone(), node);
+
+        let mut changes = HashMap::new();
+        changes.insert(key.clone(), Vec::new());
+        for location in locations {
+            let change = lsp::TextEdit {
+                range: location.range,
+                new_text: params.new_name.clone(),
+            };
+
+            changes.get_mut(&key).unwrap().push(change);
```

Comment:
Same here. This unwrap seems unsafe.

Reply:
Normally, yes, this would be unsafe, but I explicitly added an entry for this key a few lines above, so I would not expect this to panic.

Reply:
Can we refactor this then? I think this might be better.

```
let edits = locations.map(|location| lsp::TextEdit {
    range: location.range,
    new_text: params.new_name.clone(),
});

let mut changes = HashMap::new();
changes.insert(key.clone(), edits);

let response = lsp::WorkspaceEdit {
    changes: Some(changes),
    document_changes: None,
    change_annotations: None,
};
Ok(Some(response))
```